libs/wingsdk/src/cloud/function.ts

```diff
@@ -100,14 +100,14 @@ export class Function extends Resource implements IInflightHost {
   public _preSynthesize(): void {
     super._preSynthesize();
 
-    // write the entrypoint next to the partial inflight code emitted by the compiler, so that
-    // `require` resolves naturally.
+    // write the entrypoint next to the partial inflight code emitted by the compiler,
+    // so that `require` resolves naturally.
     const lines = this._getCodeLines(this.handler);
     writeFileSync(this.entrypoint, lines.join("\n"));
 
     // indicates that we are calling the inflight constructor and the
     // inflight "handle" method on the handler resource.
-    this.handler.onLift(this, ["handle", "$inflight_init"]);
+    Lifting.lift(this.handler, this, ["handle"]);
```

Comment:
(half baked thoughts) Eventually I imagine this lifting thing will need to be a public API. But right now it's called in this `_preSynthesize` method, and based on past issues dealing with `preSynthesize` bugs in projen, I'd like to avoid exposing the capability for users to execute arbitrary code in this lifecycle hook.

Some possible solutions (not planning to implement in this PR) could be:
1. Change the API so that `Lifting.lift` can be called inside your class constructor, but it only does its actual work during `preSynthesize` under the hood
2. Add a special base class for compute resources with some built-in methods you can use, e.g.

```js
bring cloud;

class Function extends std.InflightHost {
  new() {
    // ...
    super.lift(this.handler, ["handle"]);
    // ...
  }
}
```

Anyway I think we have a lot of options for how to handle this